### PR TITLE
CI: add py38 slow build #35160

### LINKED ISSUE #35160: CI: add sql tests back to py36
xref https://github.com/pandas-dev/pandas/issues/34776

removed in the PR: https://github.com/pandas-dev/pandas/pull/35151, we should add them back.

cc @VirosaLi

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #35160

This issue was created before we dropped py36 support. So I think we should add equivalent CI build for py38.

### CODE REVIEW

**Comment by jreback** [comment]
can u out this on azure instead? we have more parallelism there - and they are slightly less opaque than travis

**Comment by fangchenli** [comment]
> can u out this on azure instead? we have more parallelism there - and they are slightly less opaque than travis

I'll look at how to set up sql on azure. It's more straightforward to do this on Travis.

**Comment by jreback** [comment]
@fangchenli another alternative is to move one of the other travis builds to azure if that's easier.

**Comment by jreback** [comment]
can you rebase; also is the 3.7_locale duplicated in azure & travis? (or is it different)?

**Comment by fangchenli** [comment]
> can you rebase; also is the 3.7_locale duplicated in azure & travis? (or is it different)?

The one on travis has sql related tests. Maybe we can combine those two.

File: ci/deps/travis-37-locale.yaml [hunk 1]
```diff
@@ -20,22 +25,23 @@ dependencies:
...
-  - python-dateutil
-  - pytz
   - scipy
-  - sqlalchemy=1.3.0
   - xarray=0.12.0
   - xlrd
   - xlsxwriter
   - xlwt
+  - moto
+  - flask
```
Comment (fangchenli) [inline ci/deps/travis-37-locale.yaml:42]: Those two were in the auzre-37-locale build.

File: ci/azure/posix.yml [hunk 2]
```diff
@@ -20,39 +20,35 @@ jobs:
...
-          PATTERN: "not slow and not network"
-          LANG: "it_IT.utf8"
-          LC_ALL: "it_IT.utf8"
-          EXTRA_APT: "language-pack-it xsel"
-
-#        py37_32bit:
-#          ENV_FILE: ci/deps/azure-37-32bit.yaml
-#          CONDA_PY: "37"
-#          PATTERN: "not slow and not network and not clipboard"
-#          BITS32: "yes"
```
Comment (fangchenli) [inline ci/azure/posix.yml:50]: Conda doesn't have 32bit build anymore.

File: .travis.yml [hunk 3]
```diff
@@ -41,10 +41,10 @@ matrix:
         - JOB="3.9-dev" PATTERN="(not slow and not network and not clipboard)"
 
     - env:
-        - JOB="3.8" ENV_FILE="ci/deps/travis-38.yaml" PATTERN="(not slow and not network and not clipboard)"
-
-    - env:
-        - JOB="3.7" ENV_FILE="ci/deps/travis-37.yaml" PATTERN="(not slow and not network and not clipboard)"
```
Comment (fangchenli) [inline .travis.yml:47]: Those two don't use any travis services.

**Comment by jreback** [comment]
ok this lgtm. can you merge master to retrigger and ping on green.

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**